test/language/statements/for-of/iterator-next-reference.js

```diff
@@ -3,39 +3,33 @@
 /*---
 es6id: 13.6.4.13 S5.c
 description: >
-    The iterator's `next` method should be accessed with each iteration as per
-    the `IteratorStep` abstract operation (7.4.5).
+    The iterator's `next` method should be accessed only once with each
+    iteration as per the `GetIterator` abstract operation (7.4.1).
 features: [Symbol.iterator]
 ---*/
 
 var iterable = {};
 var iterator = {};
 var firstIterResult = { done: false };
-var iterationCount, invocationCount;
+var iterationCount = 0;
 
 iterable[Symbol.iterator] = function() {
   return iterator;
 };
 
-iterator.next = function() { return { value: 45, done: false }; };
-iterationCount = 0;
-invocationCount = 0;
+iterator.next = function() {
+  if (iterationCount) return { done: true };
+  return { value: 45, done: false };
+};
 for (var x of iterable) {
   assert.sameValue(x, 45);
 
-  iterator.next = function() {
-    invocationCount++;
-
-    Object.defineProperty(iterator, 'next', {
-      get: function() {
-        $ERROR('Should not access the `next` method after iteration ' +
-          'is complete.');
-      }
-    });
-
-    return { value: null, done: true };
-  };
+  Object.defineProperty(iterator, 'next', {
+    get: function() {
+      $ERROR('Should not access the `next` method after the iteration ' +
```

Comment:
while you're here, would you mind to replace this `$ERROR` call to a `throw new Test262Error`?

This should not be a blocker.

Reply:
✔️